armi/utils/plotting.py

```diff
@@ -1047,3 +1063,304 @@ def makeHistogram(x, y):
+    """
+    from armi.reactor.components import DerivedShape
+
+    patches = []
+    data = []
+    name = []
+    if isinstance(block.spatialGrid, grids.HexGrid):
+        largestPitch, comp = block.getPitch(returnComp=True)
+
+    elif isinstance(block.spatialGrid, grids.ThetaRZGrid):
+        raise TypeError(
+            "This plot function is not currently supported for ThetaRZGrid grids."
+        )
+    else:
+        largestPitch, comp = block.getPitch(returnComp=True)
+        if block.getPitch()[0] != block.getPitch()[1]:
+            raise ValueError("Only works for blocks with equal length and width.")
+
+    sortedComp = [c for c in block]
+
+    derivedComponent = block.getComponentsOfShape(DerivedShape)
+    if len(derivedComponent) == 1:
+        component = derivedComponent[0]
+        cName = component.name
+        material = component.material.name
+        if isinstance(comp, Hexagon):
+            derivedPatch = matplotlib.patches.RegularPolygon(
+                (0, 0), 6, largestPitch / math.sqrt(3)
+            )
+        elif isinstance(comp, Square):
+            derivedPatch = matplotlib.patches.Rectangle(
+                (0, 0), largestPitch[0], largestPitch[0]
+            )
+        else:
+            raise TypeError(
+                "Shape of the pitch-defining element is not a Square or Hex it is {}, cannot plot for this type of block".format(
+                    comp.shape
+                )
+            )
+        patches.append(derivedPatch)
+        data.append(material)
+        name.append(cName)
+    sortedComp.sort(reverse=True)
+    for components in sortedComp:
+        if components.name is not "pitch" and not isinstance(components, DerivedShape):
+            loc = components.spatialLocator
+            if not isinstance(loc, grids.MultiIndexLocation):
+                # make a single location a list to iterate.
+                loc = [
+                    loc,
+                ]
+            for local in loc:
+                x, y, _ = local.getLocalCoordinates()
+                # goes through each location
+                # want to place a patch at that location
+                blockPatch = _makeComponentPatch(components, (x, y), cold)
+                for element in blockPatch:
+                    patches.append(element)
+                    data.append(components.material.name)
+                    name.append(components.name)
+
+    return patches, data, name
+
+
+def _makeComponentPatch(component, position, cold):
+    """Makes a component shaped patch to later be used for making block diagrams.
+
+    Parameters
+    ----------
+        component: a component of a block
+
+        position: tuple
+            (x, y) position
+
+        cold: boolean
+            True if looking for dimension at cold temps
+
+    Return
+    ------
+        blockPatch: List
+            A list that is either one element (a single patch), or 6 in the case of a Hexagon
```

Comment:
Probably no need to go to that much detail about how many patches; just say something like "a list of Patch objects representing the component in the diagram."